File: practice/algorithms/sorting/closest_numbers/closest_numbers.rb
tests = gets.to_i
tests.times do
  arr = gets.to_s.split.map{|s| s.to_i}.sort{|a, b| a <=> b}
  min_diff = 100000000;
  for i in 0...(arr.size - 1)
    diff = arr[i + 1] - arr[i]
    min_diff = diff if(diff < min_diff)
  end
  elements = []
  for i in 0...(arr.size - 1)
    diff = arr[i + 1] - arr[i]
    if(diff == min_diff)
      elements << arr[i]
      elements << arr[i + 1]
    end
  end
  elements.sort!{|a, b| a <=> b}
  puts elements.join(" ")
end

File: practice/algorithms/sorting/closest_numbers/ClosestNumbers.js
'use strict';

const processData = input => {
    let lines = input.split('\n');
    let array = lines[1].split(' ').map(i => parseInt(i)).sort((a, b) => a - b);
    let minDiff = 1000000000000000000;
    for(let i = 0; i < array.length - 1; i++) {
        let diff = array[i + 1] - array[i];
        if(diff < minDiff) {
            minDiff = diff;
        }
    }
    let elements = [];
    for(let i = 0; i < array.length - 1; i++) {
        let diff = array[i + 1] - array[i];
        if(diff == minDiff) {
            elements.push(array[i]);
            elements.push(array[i + 1]);
        }
    }
    elements = elements.sort((a, b) => a - b);
    console.log(elements.join(" "));
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));